test/e2e/framework.go

```diff
@@ -2906,6 +2920,23 @@ func (data *TestData) createAgnhostPodOnNode(name string, ns string, nodeName st
 	return NewPodBuilder(name, ns, agnhostImage).OnNode(nodeName).WithHostNetwork(hostNetwork).Create(data)
 }
 
+func (data *TestData) createAgnhostPodWithHTTPOnNode(name string, ns string, nodeName string, hostNetwork bool) error {
+	args := []string{"netexec", "--http-port=8080"}
+	ports := []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: 8080,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+	return NewPodBuilder(name, ns, agnhostImage).
+		OnNode(nodeName).
+		WithArgs(args).
+		WithPorts(ports).
+		WithHostNetwork(hostNetwork).
+		Create(data)
+}
```

Comment:
please don't define new specialized helper functions like this one, we have too many already :/
Instead use `NewPodBuilder` directly in the test